protocol/xmrtaker/instance.go

```diff
@@ -282,7 +282,7 @@ func (inst *Instance) maybeCancelNewSwap(txHash ethcommon.Hash) (bool, error) {
 	// loop for 1 hour and block as there will never be a receipt for the cancel tx.
 	// we should probably poll for our account nonce to increase, and when it does, check for receipts
 	// on both txs to see which one was successful.
-	receipt, err := block.WaitForReceipt(inst.backend.Ctx(), inst.backend.ETHClient().Raw(), cancelTx)
+	receipt, err := block.WaitForReceipt(inst.backend.Ctx(), inst.backend.ETHClient().Raw(), cancelTx.TxHash)
```

Comment:
`CancelTxWithNonce` now returns the receipt so this is unneeded, but the comment still applies, so leave the error check under this line for the receipt